### DISCUSSION TITLE
Logging - where to look

### DISCUSSION THREAD

**Discussion by richard-lee-863** [HiEventsDev/Hi.Events#188] [Q&A]
Hi, I'm super impressed with initial impressions of Hi.Events! I managed to get some test events up and running pretty quickly, but I'm struggling to diagnose some 500 server errors. 
I'm using an "all-in-one" Docker build, with the following environment variable set:

`      - LOG_CHANNEL=stderr`

I'm not really getting anything showing up in the Docker container logs though. 
For example, I'm trying to test the email functionality, but i'm only getting logs like this in Portainer:

```
- - [03/Sep/2024:12:12:23 +0000] "POST /api/events/2/attendees/1/resend-ticket HTTP/1.1" 500 44 "http://REDACTED:8123/manage/event/2/attendees" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:129.0) Gecko/20100101 Firefox/129.0"

127.0.0.1 -  03/Sep/2024:12:12:19 +0000 "POST /index.php/events/2/attendees/1/resend-ticket" 500
```

How can I actually determine what is going wrong?

**Comment by daveearley** [comment]
Hi @richard-lee-863,

Glad you like Hi.Events! 

In this case, where the backend returns a 500, the error you're looking for would usually appear above the logs you've posted. 

If you don't see any other errors, try setting `APP_DEBUG=true` as an environment variable.

As the error is on the `resend-ticket` endpoint, I would assume it's related to mail setting not being configured correctly.

Let me know what you find.

Cheers,
Dave

**Comment by richard-lee-863** [comment]
Hi Dave,

I'm 99% sure its an issue with my MAIL setup, but I did try the same setup in an smtp testing website and it worked. 
I'm trying to use Google smtp, which i've setup on a few other self hosted services using App passwords. 
I notice the app passwords have spaces in them, which I've tried to deal with by keeping them in quotes.

`MAIL_PASSWORD='fake app pass key'`

Is there anything obvious I might have overlooked apart from that?

**Comment by richard-lee-863** [comment]
Annoyingly, any logs output seem to be concatenated? I've used Portainer log viewer as well as Unraid docker log viewer, and when i got an error it would only show from about line #14 onwards which didn't help..

Anyway, managed to get the mail server working using Brevo smtp relay instead of gmail, so that will get me going for now to just test things out!
